src/monattack.cpp

```diff
@@ -3118,8 +3157,10 @@ void mattack::upgrade(monster *z, int index)
 {
     std::vector<int> targets;
     for (size_t i = 0; i < g->num_zombies(); i++) {
-        if (g->zombie(i).type->id == "mon_zombie" &&
-            rl_dist( z->pos(), g->zombie(i).pos() ) <= 5) {
+        monster &zed = g->zombie(i);
+        if( zed.type->id == "mon_zombie" &&
+            rl_dist( z->pos(), zed.pos() ) <= 5 &&
+            z->attitude_to( zed ) != Creature::Attitude::A_HOSTILE ) {
```

Comment:
OK, so if I read this right it won't upgrade zeds that aren't on its side (yours or Team Zombie, depending).  Good catch.